packages/client/src/schedule-types.ts

```diff
@@ -0,0 +1,830 @@
+   * minute, which means that the Schedule attempts to take any Actions that wouldn't be more than one minute late. It
+   * takes those Actions according to the {@link ScheduleOverlapPolicy}. An outage that lasts longer than the Catchup
+   * Window could lead to missed Actions. (But you can always {@link ScheduleHandle.backfill}.)
+   *
+   * @default 1 minute
+   * @format number of milliseconds
+   */
+  catchupWindow: number | undefined;
+
+  /**
+   * When an Action times out or reaches the end of its Retry Policy, {@link pause}.
+   *
+   * With {@link ScheduleOverlapPolicy.ALLOW_ALL}, this pause might not apply to the next Action, because the next Action
+   * might have already started previous to the failed one finishing. Pausing applies only to Actions that are scheduled
+   * to start after the failed one finishes.
+   *
+   * @default false
+   */
+  pauseOnFailure: boolean;
+
+  /**
+   * The Actions remaining in this Schedule. Once this number hits `0`, no further Actions are taken (unless {@link
+   * ScheduleHandle.trigger} is called).
+   *
+   * @default undefined (unlimited)
+   */
+  remainingActions?: number;
+
+  /** Number of Actions taken so far. */
+  numActionsTaken: number;
+
+  /** Number of times a scheduled Action was skipped due to missing the catchup window. */
+  numActionsMissedCatchupWindow: number;
+
+  /** Number of Actions skipped due to overlap. */
+  numActionsSkippedOverlap: number;
+
+  /**
+   * Currently-running workflows started by this schedule. (There might be
+   * more than one if the overlap policy allows overlaps.)
+   */
+  runningWorkflows: WorkflowExecutionWithFirstExecutionRunId[];
+
+  createdAt: Date;
+  lastUpdatedAt: Date | undefined;
+
+  raw: temporal.api.workflowservice.v1.IDescribeScheduleResponse;
+};
+
+export type UpdatedSchedule<Action extends Workflow> = Pick<
+  ScheduleOptions<Action>,
+  'spec' | 'action' | 'overlap' | 'catchupWindow' | 'pauseOnFailure' | 'note' | 'paused'
+>;
+
+/**
+ * Make all properties optional.
+ *
+ * If original Schedule is deleted, okay to use same `id`.
+ */
+export type ScheduleOptionsOverrides<Action extends Workflow> = Partial<ScheduleOptions<Action>>;
+
+export interface WorkflowExecutionWithFirstExecutionRunId {
+  workflowId: string;
+
+  /**
+   * The Run Id of the original execution that was started by the Schedule. If the Workflow retried, did
+   * Continue-As-New, or was Reset, the following runs would have different Run Ids.
+   */
+  firstExecutionRunId: string;
+}
+
+export interface ScheduleAction {
+  /** Time that the Action was scheduled for, including jitter. */
+  scheduledAt: Date;
+
+  /** Time that the Action was actually taken. */
+  takenAt: Date;
+
+  /** If action was {@link StartWorkflowAction}. */
+  workflow?: WorkflowExecutionWithFirstExecutionRunId;
+}
+
+/**
+ * Policy for overlapping Actions.
+ */
+export enum ScheduleOverlapPolicy {
```

Comment:
We should probably export the original proto enums here to be consistent with the rest of the SDK.

Reply:
That will definitely make things much more easier.

I suppose that the proposal used this form instead to provide simpler names (ie. no 'SCHEDULE_OVERLAP_POLICY_*' prefixes) and user documentation.

@lorensr Do you want to chime in before I proceed with Roey's suggestion?

Reply:
I prefer ease of use over consistency here, don't think latter is that impt